pyop2/compilation.py

```diff
@@ -266,6 +267,12 @@ def get_so(self, src, extension):
                     os.rename(tmpname, soname)
             # Wait for compilation to complete
             self.comm.barrier()
+            if not os.path.isfile(soname):
+                # Wait more if .so file is not visible on disk
+                for i in range(60):
+                    sleep(1.0)
+                    if os.path.isfile(soname):
+                        break
```

Comment:
I wonder if there is portable way that does not involve polling... some timed out waiting for change in the directory or anything similar.

Reply:
Doubtful, you might think something like inotify would work, but that wouldn't be supported by NFS, since the place where the write happens is not the same kernel as the one where the notification needs to take place.

This is more generally motivation to resurrect the node-local, or process-local, compilation, which I will try and do in october.